sdk/communication/AzureCommunicationCommon/Source/CommunicationIdentifierFactory.h

```diff
@@ -0,0 +1,35 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+#ifndef CommunicationIdentifierFactory_h
+#define CommunicationIdentifierFactory_h
+@class CommunicationIdentifier;
+
+@interface CommunicationIdentifierFactory : NSObject
++ (id<CommunicationIdentifier>)createCommunicationIdentifier: (NSString *) rawId NS_SWIFT_UNAVAILABLE("Exposed in Swift as createCommunicationIdentifier(from: ) -> CommunicationIdentifier");
```

Comment:
If we went the route of having a separate ObjC/Swift implementation, why is the Swift implementation not the one that does not require casting? Did we engage the archboard on the range of options?

Reply:
I'm not sure I understand the question. Right now both implementations return the base protocol `CommunicationIdentifier` so they both require casting, as per our discussions around approach 2. 
The only swift implementation that didn't require casting I believe was having the associated value in the enum, but we were worried about the potential for breaking changes. 
The arch board is being engaged, with the git issue created here: https://github.com/Azure/azure-sdk/issues/4473

Reply:
If you've got the notes in a gist or whatever, I can bring it up to them and get their initial thoughts. To me that seems better than merging one implementation and then having them turn around and say "change it all!"